extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -483,12 +483,15 @@ export class AmpA4A extends AMP.BaseElement {
     }
     // If in localDev `type=fake` Ad specifies `force3p`, it will be forced
     // to go via 3p.
-    if (getMode().localDev &&
-        this.element.getAttribute('type') == 'fake' &&
-        this.element.getAttribute('force3p') == 'true') {
-      this.adUrl_ = this.getAdUrl();
-      this.adPromise_ = Promise.resolve();
-      return;
+    if (getMode().localDev && this.element.getAttribute('type') == 'fake') {
+      // Prevent issuing a second network request to try to fetch
+      // the same JSON again.
+      this.experimentalNonAmpCreativeRenderMethod_ = XORIGIN_MODE.NAMEFRAME;
```

Comment:
My original approach was to try to set it from AmpAdNetworkFakeImpl, but it's private to the parent class. I agree this is hacky, but I think any other approach would also be hacky. We could have AmpAdNetworkFakeImpl set some attribute, and then have the parent class say if (this.theAttribute) { this.experimentalNonAmpCreativeRenderMethod_ = theAttribute; } but I'm really bothered by doing any subclass-specific logic in the parent class...